js/app/modules/banner/dynamic.js

```diff
@@ -44,6 +44,13 @@ var Dynamic = new Module.Class({
          */
         'mode': GObject.ParamSpec.string('mode', 'mode', 'mode',
             GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT, 'text'),
+         /**
+         * Property: Layout
+         *
+         * The layout of the logo in full mode, it can be auto, horizontal or vertical.
+         */
+        'layout': GObject.ParamSpec.string('layout', 'layout', 'layout',
```

Comment:
Drive-by comment: it would be good to have an enum here; for the YAML file it makes no difference.

Reply:
As we spoke, we don't do this anywhere and seems it can't be done (without creating a C enum I supposed).